File: animal.js
class Animal {
    constructor(height, weight) {
        this.height = height;
        this.weight = weight;
        this.hunger = 0;
        this.thirst = 0;
    };
    grow(amount) {
        this.height += amount;
    };
    fatten(amount) {
        this.weight += amount;
    };
    drink() {
        this.thirst -= 10;
    }
    eat() {
        this.hunger -= 10;
        this.fatten(3);
    };
    run(distance) {
        let weightLost = (distance * 0.2);
        this.fatten(-weightLost)
        this.hunger += distance * 0.5;
        this.thirst += distance * 0.5;
    };
};

module.exports = Animal;

File: csharp/animal.cs
using System;

namespace csharp
{
    class Animal 
    {
        public double Height;
        public double Weight;
        public double Hunger = 0;
        public double Thirst = 0;

        public Animal(double height, double weight) 
        {
            this.Height = height;
            this.Weight = weight;
        }
        public void grow(double amount)
        {
            this.Height += amount;
        }
        public void fatten(double amount)
        {
            this.Weight += amount;
        }
        public void drink()
        {
            this.Thirst -= 10;
        }
        public void eat()
        {
            this.Hunger -= 10;
            this.fatten(3);
        }
        public void run(double distance)
        {
            double weightLost = (distance * 0.2);
            this.fatten(-weightLost);
            this.Hunger += distance * 0.5;
            this.Thirst += distance * 0.5;
        }
    }
}